packages/app-builder/src/routes/_builder+/decisions+/$decisionId.tsx

```diff
@@ -56,26 +54,39 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
 
   try {
     const currentDecision = await decision.getDecisionById(decisionId);
-    const scenarioId = currentDecision.scenario.id;
-
-    const operatorsPromise = editor.listOperators({
-      scenarioId,
-    });
-
-    const accessorsPromise = editor.listAccessors({
-      scenarioId,
-    });
 
-    const dataModelPromise = apiClient.getDataModel();
-    const { custom_lists } = await apiClient.listCustomLists();
+    const astRuleData = Promise.all([
+      scenario.getScenarioIteration({
+        iterationId: currentDecision.scenario.scenarioIterationId,
+      }),
+      editor.listOperators({
+        scenarioId: currentDecision.scenario.id,
+      }),
+      editor.listAccessors({
+        scenarioId: currentDecision.scenario.id,
+      }),
+      dataModelRepository.getDataModel(),
+      apiClient.listCustomLists(),
+    ]).then(
+      ([
+        scenarioIteration,
+        astOperators,
+        accessors,
+        dataModel,
+        { custom_lists },
+      ]) => ({
+        rules: scenarioIteration.rules,
+        databaseAccessors: accessors.databaseAccessors,
+        payloadAccessors: accessors.payloadAccessors,
+        astOperators,
+        dataModel,
+        customLists: custom_lists,
+      }),
+    );
 
-    return json({
+    return defer({
```

Comment:
Key part of the new commits :
- fetch additional data related to the current decision 
- use `defer` so it do not block the navigation (= start fetching additional data, render with available data and display a loader until data is streamed to the client)

Reply:
oh, this is surprising syntax. So using "defer" here protects us against uncaught exception and the like even without using await or catch above ?

Reply:
(I mean ok not that surprising but I didn't know this)